airflow/models/taskinstance.py

```diff
@@ -371,6 +373,802 @@ def _creator_note(val):
         return TaskInstanceNote(*val)
 
 
+def _execute_task(task_instance, context, task_orig):
```

Comment:
Let\s add typing here: And this is going to be TaskInstace | TaskInstancePydantic as this one runs on the client side.

Reply:
Yes. I think we can have few extra passes to add those typings later.